src/physics/arcade/World.js

```diff
@@ -1354,7 +1356,16 @@ var World = new Class({
         velocityX = Clamp(velocityX, -maxX, maxX);
         velocityY = Clamp(velocityY, -maxY, maxY);
 
-        body.velocity.set(velocityX, velocityY);
+        newSpeed = Math.sqrt(velocityX * velocityX + velocityY * velocityY);
```

Comment:
This calculation should be skipped if maxSpeed < 0.

Reply:
Thank you for opinion. I will apply it.